client/components/place/place-list/place-list.component.js

```diff
@@ -79,6 +80,222 @@ angular.module('placeList', ['filterMapType', 'popularTracks', 'ngAnimate'])
         };
         // -----END ADD Place-----
 
+        // -----START ADD Track-----
+        ctrl.newTrackObject = angular.copy(constants.emptyTrackModel);
+        ctrl.addPointMenuIsOpen = false;
+        ctrl.addTrackMenuIsOpen = false;
+        ctrl.newPoint = angular.copy(constants.emptyPlaceModel);
+        var newTrack;
+        var newPointForTrack;
+        var placesForTrack = [];
+        var newPointsForTrack = [];
+        ctrl.newTrackPoints = [];
+
+        ctrl.toggleAddTrackMenu = function(form) {
+          map = placesOnMap.map;
+          if (ctrl.addTrackMenuIsOpen) {
+            ctrl.addTrackMenuIsOpen = false;
+            ctrl.resetAddTrackForm(form);
+            map.off('click', addNewTrackPointOnMap);
+            for (var key in placesOnMap.places) {
+              placesOnMap.places[key].forEach(function(place) {
+                place.off('click', addExistingPointIntoNewTrack);
+              });
+            }
+          } else {
+            ctrl.addTrackMenuIsOpen = true;
+            ctrl.addPlaceMenuIsOpen = false;
+            map.on('click', addNewTrackPointOnMap);
+            for (var key2 in placesOnMap.places) {
+              placesOnMap.places[key2].forEach(function(place) {
+                place.on('click', addExistingPointIntoNewTrack);
+              });
+            }
+          }
+        };
+
+        function addExistingPointIntoNewTrack() {
+          map = placesOnMap.map;
+          var existingPoint = {
+            name: '',
+            _id: '',
+            location: {
+              coordinates: []
+            }
+          };
+          placesForTrack.push([this._latlng.lat, this._latlng.lng]);
+          existingPoint.name = this.name;
```

Comment:
Because here this is reference to object which returns when user clicks on existing marker (creating new track).